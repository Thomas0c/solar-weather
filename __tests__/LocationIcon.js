import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import LocationIcon from '../lib/js/app/components/locationIcon';

test('renders correctly when selected', () => {
  const tree = renderer.create(<LocationIcon selected name="Tokyo" />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly when not selected', () => {
  const tree = renderer.create(<LocationIcon selected={false} name="Tokyo" />).toJSON();
  expect(tree).toMatchSnapshot();
});
