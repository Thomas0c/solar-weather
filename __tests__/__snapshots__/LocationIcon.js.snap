// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when not selected 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityLiveRegion={undefined}
  accessibilityTraits={undefined}
  accessibilityViewIsModal={undefined}
  accessible={undefined}
  collapsable={undefined}
  hitSlop={undefined}
  importantForAccessibility={undefined}
  needsOffscreenAlphaCompositing={undefined}
  onAccessibilityTap={undefined}
  onLayout={undefined}
  onMagicTap={undefined}
  pointerEvents={undefined}
  removeClippedSubviews={undefined}
  renderToHardwareTextureAndroid={undefined}
  shouldRasterizeIOS={undefined}
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "height": "100.00%",
        "justifyContent": "center",
        "width": "100.00%",
      },
      Object {
        "backgroundColor": "#EFEFEF",
      },
    ]
  }
  testID={undefined}
>
  <Text
    accessible={true}
    adjustsFontSizeToFit={undefined}
    allowFontScaling={true}
    ellipsizeMode="tail"
    minimumFontScale={undefined}
    numberOfLines={undefined}
    onLayout={undefined}
    onLongPress={undefined}
    onPress={undefined}
    pressRetentionOffset={undefined}
    selectable={undefined}
    selectionColor={undefined}
    style={
      Array [
        Object {
          "fontFamily": "Baskerville",
          "fontSize": 16,
        },
        Object {
          "color": "#343434",
        },
      ]
    }
    suppressHighlighting={undefined}
    testID={undefined}
    textBreakStrategy={undefined}
  >
    Tokyo
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityLiveRegion={undefined}
    accessibilityTraits={undefined}
    accessibilityViewIsModal={undefined}
    accessible={undefined}
    collapsable={undefined}
    hitSlop={undefined}
    importantForAccessibility={undefined}
    needsOffscreenAlphaCompositing={undefined}
    onAccessibilityTap={undefined}
    onLayout={undefined}
    onMagicTap={undefined}
    pointerEvents={undefined}
    removeClippedSubviews={undefined}
    renderToHardwareTextureAndroid={undefined}
    shouldRasterizeIOS={undefined}
    style={
      Object {
        "alignItems": "center",
        "height": "50.00%",
        "justifyContent": "center",
        "width": "50.00%",
      }
    }
    testID={undefined}
  >
    <Image
      accessibilityLabel={undefined}
      accessible={undefined}
      blurRadius={undefined}
      capInsets={undefined}
      defaultSource={undefined}
      onError={undefined}
      onLayout={undefined}
      onLoad={undefined}
      onLoadEnd={undefined}
      onLoadStart={undefined}
      onPartialLoad={undefined}
      onProgress={undefined}
      resizeMethod={undefined}
      resizeMode={undefined}
      source={
        Object {
          "testUri": "../../../assets/weather_icons/sunny.png",
        }
      }
      style={
        Object {
          "alignSelf": "center",
          "marginTop": "2.00%",
          "resizeMode": "contain",
          "width": "55.00%",
        }
      }
      testID={undefined}
    />
  </View>
</View>
`;

exports[`renders correctly when selected 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityLiveRegion={undefined}
  accessibilityTraits={undefined}
  accessibilityViewIsModal={undefined}
  accessible={undefined}
  collapsable={undefined}
  hitSlop={undefined}
  importantForAccessibility={undefined}
  needsOffscreenAlphaCompositing={undefined}
  onAccessibilityTap={undefined}
  onLayout={undefined}
  onMagicTap={undefined}
  pointerEvents={undefined}
  removeClippedSubviews={undefined}
  renderToHardwareTextureAndroid={undefined}
  shouldRasterizeIOS={undefined}
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "height": "100.00%",
        "justifyContent": "center",
        "width": "100.00%",
      },
      Object {
        "backgroundColor": "#adb3b4",
      },
    ]
  }
  testID={undefined}
>
  <Text
    accessible={true}
    adjustsFontSizeToFit={undefined}
    allowFontScaling={true}
    ellipsizeMode="tail"
    minimumFontScale={undefined}
    numberOfLines={undefined}
    onLayout={undefined}
    onLongPress={undefined}
    onPress={undefined}
    pressRetentionOffset={undefined}
    selectable={undefined}
    selectionColor={undefined}
    style={
      Array [
        Object {
          "fontFamily": "Baskerville",
          "fontSize": 16,
        },
        Object {
          "color": "#FFF",
        },
      ]
    }
    suppressHighlighting={undefined}
    testID={undefined}
    textBreakStrategy={undefined}
  >
    Tokyo
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityLiveRegion={undefined}
    accessibilityTraits={undefined}
    accessibilityViewIsModal={undefined}
    accessible={undefined}
    collapsable={undefined}
    hitSlop={undefined}
    importantForAccessibility={undefined}
    needsOffscreenAlphaCompositing={undefined}
    onAccessibilityTap={undefined}
    onLayout={undefined}
    onMagicTap={undefined}
    pointerEvents={undefined}
    removeClippedSubviews={undefined}
    renderToHardwareTextureAndroid={undefined}
    shouldRasterizeIOS={undefined}
    style={
      Object {
        "alignItems": "center",
        "height": "50.00%",
        "justifyContent": "center",
        "width": "50.00%",
      }
    }
    testID={undefined}
  >
    <Image
      accessibilityLabel={undefined}
      accessible={undefined}
      blurRadius={undefined}
      capInsets={undefined}
      defaultSource={undefined}
      onError={undefined}
      onLayout={undefined}
      onLoad={undefined}
      onLoadEnd={undefined}
      onLoadStart={undefined}
      onPartialLoad={undefined}
      onProgress={undefined}
      resizeMethod={undefined}
      resizeMode={undefined}
      source={
        Object {
          "testUri": "../../../assets/weather_icons/sunny.png",
        }
      }
      style={
        Object {
          "alignSelf": "center",
          "marginTop": "2.00%",
          "resizeMode": "contain",
          "width": "55.00%",
        }
      }
      testID={undefined}
    />
  </View>
</View>
`;
